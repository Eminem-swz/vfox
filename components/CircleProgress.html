<!DOCTYPE html>
<html lang="zh-CN">
  <head>
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width,initial-scale=1">
    <title>CircleProgress 环形进度条 | Vfox for Vue3</title>
    <meta name="description" content="移动端 Vue 组件库">
    <link rel="stylesheet" href="/vfox/assets/style.133cf44e.css">
    <link rel="modulepreload" href="/vfox/assets/Home.999026eb.js">
    <link rel="modulepreload" href="/vfox/assets/app.3c8a5e7e.js">
    <link rel="modulepreload" href="/vfox/assets/components_CircleProgress.md.26ad80db.lean.js">
    
    <meta name="twitter:title" content="CircleProgress 环形进度条 | Vfox for Vue3">
  <meta property="og:title" content="CircleProgress 环形进度条 | Vfox for Vue3">
  </head>
  <body>
    <div id="app"><!--[--><div class="theme"><header class="nav-bar" data-v-3eed0030><div class="sidebar-button" data-v-3eed0030><svg class="icon" xmlns="http://www.w3.org/2000/svg" aria-hidden="true" role="img" viewBox="0 0 448 512"><path fill="currentColor" d="M436 124H12c-6.627 0-12-5.373-12-12V80c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12zm0 160H12c-6.627 0-12-5.373-12-12v-32c0-6.627 5.373-12 12-12h424c6.627 0 12 5.373 12 12v32c0 6.627-5.373 12-12 12z" class></path></svg></div><a class="nav-bar-title" href="/vfox/" aria-label="Vfox for Vue3, back to home" data-v-3eed0030 data-v-d91ad9e0><!----> Vfox for Vue3</a><div class="flex-grow" data-v-3eed0030></div><div class="nav" data-v-3eed0030><nav class="nav-links" data-v-3eed0030 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/playground/" target="_blank" data-v-a00c0156>Playground <!----></a></div></div><!--]--><!----><!----></nav></div><!--[--><!--]--></header><aside class="sidebar" data-v-1314d532><nav class="nav-links nav" data-v-1314d532 data-v-65a69246><!--[--><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/" data-v-a00c0156>指南 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/design/" data-v-a00c0156>设计 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item active" href="/vfox/components/Button" data-v-a00c0156>组件 <!----></a></div></div><div class="item" data-v-65a69246><div class="nav-link" data-v-65a69246 data-v-a00c0156><a class="item" href="/vfox/playground/" target="_blank" data-v-a00c0156>Playground <!----></a></div></div><!--]--><!----><!----></nav><!--[--><!--]--><ul class="sidebar-links" data-v-1314d532><!--[--><li class="sidebar-link"><p class="sidebar-link-item">基础</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Button">Button</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Icon">Icon</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Image">Image</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollView">ScrollView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Drawer">Drawer</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Modal">Modal</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Popover">Popover</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dropdown">Dropdown</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">反馈</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Toast">Toast</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Dialog">Dialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Notify">Notify</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActionSheet">ActionSheet</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopDialog">PopDialog</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SwipeCell">SwipeCell</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">数据输入</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Form">Form</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Input">Input</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Radio">Radio</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Checkbox">Checkbox</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Slider">Slider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Switch">Switch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stepper">Stepper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cascader">Cascader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Picker">Picker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Calendar">Calendar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SearchBar">SearchBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NumberKeyboard">NumberKeyboard</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImageUploader">ImageUploader</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/DatePicker">DatePicker</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Range">Range</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Rate">Rate</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">导航</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/BackTop">BackTop</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NavBar">NavBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/PopMenu">PopMenu</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tab">Tab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/SideTab">SideTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabBar">TabBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TabView">TabView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Sticky">Sticky</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ScrollTab">ScrollTab</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/IndexView">IndexView</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Fixed">Fixed</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Pagination">Pagination</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">展示</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Layout">Layout</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Cell">Cell</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Group">Group</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Collapse">Collapse</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Empty">Empty</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Skeleton">Skeleton</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Swiper">Swiper</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ImagePreview">ImagePreview</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/FlatList">FlatList</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/NoticeBar">NoticeBar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Divider">Divider</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Badge">Badge</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Progress">Progress</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Tag">Tag</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Order">Order</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountDown">CountDown</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Steps">Steps</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Price">Price</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Avatar">Avatar</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Timeline">Timeline</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Stopwatch">Stopwatch</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Result">Result</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/LoadMore">LoadMore</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/TimeAgo">TimeAgo</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/CountUp">CountUp</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item active" href="/vfox/components/CircleProgress">CircleProgress</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#import">Import</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#props">Props</a><!----></li><li class="sidebar-link"><a class="sidebar-link-item" href="#slots">Slots</a><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="#内容区（-default）">内容区（#default）</a><!----></li></ul></li></ul></li><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/ActivityIndicator">ActivityIndicator</a><!----></li></ul></li><li class="sidebar-link"><p class="sidebar-link-item">其他</p><ul class="sidebar-links"><li class="sidebar-link"><a class="sidebar-link-item" href="/vfox/components/Copy">Copy</a><!----></li></ul></li><!--]--></ul><!--[--><!--]--></aside><div class="sidebar-mask"></div><main class="page" data-v-33252f1f><div class="container" data-v-33252f1f><!--[--><!--[--><!----><!--]--><!--[--><div class="demo"><iframe style="" src="/vfox/demo/#/CircleProgress" frameborder="0"></iframe></div><!--]--><!--]--><div style="position:relative;" class="content" data-v-33252f1f><div><h1 id="circleprogress-环形进度条" tabindex="-1">CircleProgress 环形进度条 <a class="header-anchor" href="#circleprogress-环形进度条" aria-hidden="true">#</a></h1><h2 id="import" tabindex="-1">Import <a class="header-anchor" href="#import" aria-hidden="true">#</a></h2><div class="language-JavaScript"><pre><code><span class="token keyword">import</span> <span class="token punctuation">{</span> CircleProgress <span class="token punctuation">}</span> <span class="token keyword">from</span> <span class="token string">&#39;vfox&#39;</span>
</code></pre></div><p>具体的引入方式可以参考<a href="./../guide/import.html">引入组件</a>。</p><h2 id="props" tabindex="-1">Props <a class="header-anchor" href="#props" aria-hidden="true">#</a></h2><table><thead><tr><th>属性</th><th>类型</th><th>默认值</th><th>必填</th><th>说明</th></tr></thead><tbody><tr><td>percentage</td><td>string | number</td><td></td><td>是</td><td>百分比，例如：50</td></tr><tr><td>size</td><td>string | number</td><td>100</td><td>否</td><td>宽高，单位 px</td></tr><tr><td>stroke-width</td><td>string | number</td><td>5.37</td><td>否</td><td>进度条宽度，单位 px</td></tr><tr><td>color</td><td>string</td><td></td><td>否</td><td>进度条激活色</td></tr></tbody></table><h2 id="slots" tabindex="-1">Slots <a class="header-anchor" href="#slots" aria-hidden="true">#</a></h2><h3 id="内容区（-default）" tabindex="-1">内容区（#default） <a class="header-anchor" href="#内容区（-default）" aria-hidden="true">#</a></h3><div class="language-Vue"><pre><code><span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>fx-circle-progress</span> <span class="token attr-name">:percentage</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>5<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span>
  <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;</span>template</span> <span class="token attr-name">#default</span><span class="token attr-value"><span class="token punctuation attr-equals">=</span><span class="token punctuation">&quot;</span>{ progress }<span class="token punctuation">&quot;</span></span><span class="token punctuation">&gt;</span></span> 已抢{{ progress }} <span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>template</span><span class="token punctuation">&gt;</span></span>
<span class="token tag"><span class="token tag"><span class="token punctuation">&lt;/</span>fx-circle-progress</span><span class="token punctuation">&gt;</span></span>
</code></pre></div></div></div><footer class="page-footer" data-v-33252f1f data-v-c772b490><div class="edit" data-v-c772b490><div class="edit-link" data-v-c772b490 data-v-11add020><!----></div></div><div class="updated" data-v-c772b490><!----></div></footer><div class="next-and-prev-link" data-v-33252f1f data-v-6889d424><div class="container" data-v-6889d424><div class="prev" data-v-6889d424><a class="link" href="/vfox/components/CountUp" data-v-6889d424><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-prev" data-v-6889d424><path d="M19,11H7.4l5.3-5.3c0.4-0.4,0.4-1,0-1.4s-1-0.4-1.4,0l-7,7c-0.1,0.1-0.2,0.2-0.2,0.3c-0.1,0.2-0.1,0.5,0,0.8c0.1,0.1,0.1,0.2,0.2,0.3l7,7c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3c0.4-0.4,0.4-1,0-1.4L7.4,13H19c0.6,0,1-0.4,1-1S19.6,11,19,11z"></path></svg><span class="text" data-v-6889d424>CountUp</span></a></div><div class="next" data-v-6889d424><a class="link" href="/vfox/components/ActivityIndicator" data-v-6889d424><span class="text" data-v-6889d424>ActivityIndicator</span><svg xmlns="http://www.w3.org/2000/svg" viewbox="0 0 24 24" class="icon icon-next" data-v-6889d424><path d="M19.9,12.4c0.1-0.2,0.1-0.5,0-0.8c-0.1-0.1-0.1-0.2-0.2-0.3l-7-7c-0.4-0.4-1-0.4-1.4,0s-0.4,1,0,1.4l5.3,5.3H5c-0.6,0-1,0.4-1,1s0.4,1,1,1h11.6l-5.3,5.3c-0.4,0.4-0.4,1,0,1.4c0.2,0.2,0.5,0.3,0.7,0.3s0.5-0.1,0.7-0.3l7-7C19.8,12.6,19.9,12.5,19.9,12.4z"></path></svg></a></div></div></div><!--[--><!--]--></div></main></div><!----><!--]--></div>
    <script>__VP_HASH_MAP__ = JSON.parse("{\"components_actionsheet.md\":\"333ac236\",\"components_activityindicator.md\":\"21611728\",\"components_avatar.md\":\"60472e73\",\"components_backtop.md\":\"25fbb0c9\",\"components_badge.md\":\"c635d3e5\",\"components_button.md\":\"fdd28e9a\",\"components_calendar.md\":\"8fd1703f\",\"components_cascader.md\":\"566594dc\",\"components_cell.md\":\"ac34c5cf\",\"components_checkbox.md\":\"9674a260\",\"components_circleprogress.md\":\"26ad80db\",\"components_collapse.md\":\"5fde952d\",\"components_copy.md\":\"ea6822b2\",\"components_countdown.md\":\"bd6ae8fc\",\"components_countup.md\":\"b14f17bc\",\"components_datepicker.md\":\"344b83e0\",\"components_dialog.md\":\"f8ecd331\",\"components_divider.md\":\"15483b6f\",\"components_drawer.md\":\"fe31f851\",\"components_dropdown.md\":\"b0db3890\",\"components_empty.md\":\"9213c377\",\"components_fixed.md\":\"a381fdb4\",\"components_flatlist.md\":\"8df66346\",\"components_form.md\":\"ca2700ce\",\"components_group.md\":\"12a1e581\",\"components_icon.md\":\"b4f35553\",\"components_image.md\":\"5750376f\",\"components_imagepreview.md\":\"a803031f\",\"components_imageuploader.md\":\"b69acad8\",\"components_indexview.md\":\"94acc2e1\",\"components_input.md\":\"54a41093\",\"components_layout.md\":\"25d27001\",\"components_loadmore.md\":\"a602e5aa\",\"components_modal.md\":\"ce83b357\",\"components_navbar.md\":\"e39cfba6\",\"components_noticebar.md\":\"7622afcb\",\"components_notify.md\":\"9a7a4fd2\",\"components_numberkeyboard.md\":\"b57b3e9d\",\"components_order.md\":\"cb9daa16\",\"components_pagination.md\":\"48df3597\",\"components_picker.md\":\"d9bf30c5\",\"components_popdialog.md\":\"af796c22\",\"components_popmenu.md\":\"93c24474\",\"components_popover.md\":\"479641c1\",\"components_price.md\":\"7001db89\",\"components_progress.md\":\"b6b843e7\",\"components_radio.md\":\"e382854e\",\"components_range.md\":\"f134f7a3\",\"components_rate.md\":\"b9e7549a\",\"components_result.md\":\"57e5b8d2\",\"components_scrolltab.md\":\"55805950\",\"components_scrollview.md\":\"66e68966\",\"components_searchbar.md\":\"36137938\",\"components_sidetab.md\":\"ce54ef9d\",\"components_skeleton.md\":\"8ae1a444\",\"components_slider.md\":\"04127063\",\"components_stepper.md\":\"c98ac5dc\",\"components_steps.md\":\"ef2adf14\",\"components_sticky.md\":\"708ee98c\",\"components_stopwatch.md\":\"cbb3bcb6\",\"components_swipecell.md\":\"65eff9e9\",\"components_swiper.md\":\"78f87203\",\"components_switch.md\":\"a9867703\",\"components_tab.md\":\"6b482bc9\",\"components_tabbar.md\":\"32653521\",\"components_tabview.md\":\"1aef65c8\",\"components_tag.md\":\"186a05cf\",\"components_timeago.md\":\"eb412fdb\",\"components_timeline.md\":\"418c5e9d\",\"components_toast.md\":\"e2b7d04e\",\"components_index.md\":\"4a593e6d\",\"design_color.md\":\"118c02d5\",\"design_index.md\":\"d2112ced\",\"guide_dark.md\":\"8d28c9ee\",\"guide_import.md\":\"8b91b642\",\"guide_locale.md\":\"6620383b\",\"guide_theme.md\":\"3996b976\",\"index.md\":\"b0eea499\"}")</script>
    <script type="module" async src="/vfox/assets/app.3c8a5e7e.js"></script>
    
  </body>
</html>