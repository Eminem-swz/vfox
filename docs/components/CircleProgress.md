# CircleProgress 环形进度条

## Props

| 属性         | 类型          | 默认值 | 必填 | 说明                |
| ------------ | ------------- | ------ | ---- | ------------------- |
| percentage   | number/string |        | 是   | 百分比，例如：50    |
| size         | number        | 100    | 否   | 宽高，单位 px       |
| stroke-width | boolean       | 5.37   | 否   | 进度条宽度，单位 px |
| color        | string        |        | 否   | 进度条激活色        |

## Slots

### 内容区（#default）

```
<fx-circle-progress :percentage="5">
  <template #default="{ progress }"> 已抢{{ progress }} </template>
</fx-circle-progress>
```
