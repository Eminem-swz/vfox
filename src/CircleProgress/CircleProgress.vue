<template>
  <div
    class="fx-circle-progress"
    :style="{ fontSize: Math.max(9, size * 0.17 * 0.875) + 'px' }"
  >
    <loading-svg
      class="fx-circle-progress_bar"
      :size="size"
      :rate="rate"
      :strokeWidth="strokeWidth"
      :color="color"
    />
    <div
      class="fx-circle-progress_text"
      :style="{ padding: strokeWidth + 'px' }"
    >
      <slot :progress="progress">
        {{ progress }}
      </slot>
    </div>
  </div>
</template>

<script lang="ts">
import LoadingSvg from '../ActivityIndicator/LoadingSvg.vue'
import { computed, defineComponent } from 'vue'
import { isNumeric, rangeInteger } from '@/helpers/util'

export default defineComponent({
  name: 'fx-circle-progress',
  components: { LoadingSvg },
  props: {
    percentage: {
      type: [String, Number],
      validator: (val: string | number) => {
        return (
          isNumeric(val) &&
          parseFloat(val as string) >= 0 &&
          parseFloat(val as string) <= 100
        )
      },
      default: 0,
      required: true
    },
    size: {
      type: Number,
      default: 100
    },
    strokeWidth: {
      type: Number,
      default: 5.37
    },
    color: {
      type: String,
      default: '#1890ff'
    }
  },
  setup(props) {
    return {
      rate: computed(() => {
        return rangeInteger(props.percentage, 0, 100) / 100
      }),
      progress: computed(() => {
        return rangeInteger(props.percentage, 0, 100) + '%'
      })
    }
  }
})
</script>
