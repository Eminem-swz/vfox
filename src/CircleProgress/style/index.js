import '../../LoadingIcon/style/index'
import '../index.scss'
