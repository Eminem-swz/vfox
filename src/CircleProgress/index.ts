import { App } from 'vue'
import { SFCWithInstall } from '@/helpers/types'
import CircleProgress from './CircleProgress.vue'

const _CircleProgress: SFCWithInstall<typeof CircleProgress> = Object.assign(
  CircleProgress,
  {
    install: function(app: App) {
      app.component(CircleProgress.name, CircleProgress)
    }
  }
)

export default _CircleProgress
