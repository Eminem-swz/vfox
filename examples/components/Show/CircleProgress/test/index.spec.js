/* global describe,test,expect */

import { shallowMount } from '@vue/test-utils'
import CircleProgress from '@/CircleProgress'

describe('CircleProgress', () => {
  test('should render default slot correctly', async () => {
    const wrapper = shallowMount(CircleProgress, {
      props: { percentage: 0 }
    })
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ percentage: 50 })
    expect(wrapper.html()).toMatchSnapshot()

    await wrapper.setProps({ percentage: 100 })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should render custom default slot correctly', () => {
    const wrapper = shallowMount(CircleProgress, {
      scopedSlots: {
        default: `<span>Sale {{ props.progress }}</span>`
      }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })

  test('should prop percentage=800 then progress="100%" correctly', async () => {
    const wrapper = shallowMount(CircleProgress, {
      props: { percentage: 800 }
    })

    expect(wrapper.vm.progress).toBe('100%')
  })
})
