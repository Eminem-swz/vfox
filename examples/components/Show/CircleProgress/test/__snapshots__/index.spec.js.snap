// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircleProgress should render custom default slot correctly 1`] = `
<div class="fx-circle-progress" style="font-size: 14.875px;">
  <fx-loading-svg-stub class="fx-circle-progress_bar" size="100" rate="0" strokewidth="5.37" color="#1890ff"></fx-loading-svg-stub>
  <div class="fx-circle-progress_text" style="padding: 5.37px;">0%</div>
</div>
`;

exports[`CircleProgress should render default slot correctly 1`] = `
<div class="fx-circle-progress" style="font-size: 14.875px;">
  <fx-loading-svg-stub class="fx-circle-progress_bar" size="100" rate="0" strokewidth="5.37" color="#1890ff"></fx-loading-svg-stub>
  <div class="fx-circle-progress_text" style="padding: 5.37px;">0%</div>
</div>
`;

exports[`CircleProgress should render default slot correctly 2`] = `
<div class="fx-circle-progress" style="font-size: 14.875px;">
  <fx-loading-svg-stub class="fx-circle-progress_bar" size="100" rate="0" strokewidth="5.37" color="#1890ff"></fx-loading-svg-stub>
  <div class="fx-circle-progress_text" style="padding: 5.37px;">50%</div>
</div>
`;

exports[`CircleProgress should render default slot correctly 3`] = `
<div class="fx-circle-progress" style="font-size: 14.875px;">
  <fx-loading-svg-stub class="fx-circle-progress_bar" size="100" rate="0" strokewidth="5.37" color="#1890ff"></fx-loading-svg-stub>
  <div class="fx-circle-progress_text" style="padding: 5.37px;">100%</div>
</div>
`;
